etna/ensembles/voting_ensemble.py

```diff
@@ -0,0 +1,145 @@
+            number of jobs to run in parallel
+        """
+        self._validate_pipeline_number(pipelines=pipelines)
+        self.horizon = self._get_horizon(pipelines=pipelines)
+        self.weights = self._process_weights(weights=weights, pipelines_number=len(pipelines))
+        self.pipelines = pipelines
+        self.n_jobs = n_jobs
+
+    @staticmethod
+    def _validate_pipeline_number(pipelines: List[Pipeline]):
+        """Check that given valid number of pipelines."""
+        if len(pipelines) < 2:
+            raise ValueError("At least two pipelines are expected.")
+
+    @staticmethod
+    def _get_horizon(pipelines: List[Pipeline]) -> int:
+        """Get ensemble's horizon."""
+        horizons = list(set([pipeline.horizon for pipeline in pipelines]))
+        if len(horizons) > 1:
+            raise ValueError("All the pipelines should have the same horizon.")
+        return horizons[0]
+
+    @staticmethod
+    def _process_weights(weights: Optional[Iterable[float]], pipelines_number: int) -> List[float]:
+        """Process weights: if weights are not given, set them with default values, normalize weights."""
+        if weights is None:
+            weights = [1 / pipelines_number for _ in range(pipelines_number)]
+        elif len(weights) != pipelines_number:
+            raise ValueError("Weights size should be equal to pipelines number.")
+        common_weight = sum(weights)
+        weights = [w / common_weight for w in weights]
+        return weights
+
+    @staticmethod
+    def _fit_pipeline(pipeline: Pipeline, ts: TSDataset) -> Pipeline:
+        """Fit given pipeline with ts."""
+        tslogger.log(msg=f"Start fitting {pipeline.__repr__()}.")
+        pipeline.fit(ts=ts)
+        tslogger.log(msg=f"Pipeline {pipeline.__repr__()} is fitted.")
+        return pipeline
+
+    def fit(self, ts: TSDataset) -> "VotingEnsemble":
+        """Fit pipelines in ensemble.
+
+        Parameters
+        ----------
+        ts:
+            TSDataset to fit ensemble
+
+        Returns
+        -------
+        VotingEnsemble:
+            fitted ensemble
+        """
+        self.pipelines = Parallel(n_jobs=self.n_jobs, backend="multiprocessing", verbose=11)(
+            delayed(self._fit_pipeline)(pipeline=pipeline, ts=deepcopy(ts)) for pipeline in self.pipelines
+        )
+
+    @staticmethod
+    def _forecast_pipeline(pipeline: Pipeline) -> TSDataset:
+        """Make forecast with given pipeline."""
+        tslogger.log(msg=f"Start forecasting with {pipeline.__repr__()}.")
+        forecast = pipeline.forecast()
+        tslogger.log(msg=f"Forecast is done with {pipeline.__repr__()}.")
+        return forecast
+
+    def _vote(self, forecasts: List[TSDataset]) -> TSDataset:
+        """Get average forecast."""
+        forecast_df = forecasts[0][:, :, "target"] * self.weights[0]
+        for forecast, weight in zip(forecasts[1:], self.weights[1:]):
+            forecast_df += forecast[:, :, "target"] * weight
```

Comment:
Are you sure this is a good practice?
Probably we can make this computation with only one pass